StepperEngine/src/Steps/SpendSomeTimeStep.java

```diff
@@ -0,0 +1,28 @@
+package Steps;
+
+import DataTypes.StepDataType;
+
+import java.util.ArrayList;
+
+
+public class SpendSomeTimeStep extends Step{
+    public SpendSomeTimeStep(ArrayList<StepDataType> inputs, ArrayList<StepDataType> outputs) {
+        super("TIME_TO_SPEND", true, inputs, outputs);
+    }
+    @Override
+    public void execute(){
+        Integer sleepAmount= (Integer) inputs.get(0).getData();
+        this.addLog("About to sleep for " + sleepAmount + " seconds...");
+        try {
```

Comment:
Can't Thread.Sleep fail (when sleep amont >= 0 )? And if it can, we will present a misleading message to the user "A number below zero...".

maybe it's a little bit better to first validate that sleepAmout >= 0 with a simple "if". and then using try-catch for the Thread.Sleep.

and on deeper thought - possibly (although unlikely) other things might fails as well like logging and printing. 

maybe it's safer to have something like this:?

Public void execute() {
    try {
        runStepFlow();  // this actually runs the step
    }
    catch(Exception e) { 
        setSummaryLine ("Exception: " + e.message())
        setStatus(fail).....
    }
}

and then, no matter what error happens - we give a correct message and keep running.

(These are all tired 1AM thoughts - might be wrong)